pkg/operator/controller/machineconfig/controller.go

```diff
@@ -0,0 +1,360 @@
+	return nil
+}
+
+// checkProfConf checks and ensures profiling config for defined services
+func (r *MachineconfigReconciler) checkProfConf(ctx context.Context) error {
+
+	errored := false
+	errs := fmt.Errorf("failed to check profiling configs")
+	if r.CtrlConfig.Spec.EnableCrioProfiling {
+		criomc, created, err := r.ensureCrioProfConfigExists(ctx)
+		if err != nil {
+			errored = true
+			r.Log.Error(err, "failed to enable crio profiling")
+			errs = fmt.Errorf("%w: %s", errs, err.Error())
+		}
+		if created {
+			r.EventRecorder.Eventf(r.CtrlConfig, corev1.EventTypeNormal, "CreateConfig", "successfully created crio machine config")
+			r.PrevSyncChange["crio"] = PrevSyncData{
+				action: "created",
+				config: *criomc,
+			}
+		}
+	} else {
+		deleted, err := r.ensureCrioProfConfigNotExists(ctx)
+		if err != nil {
+			errored = true
+			r.Log.Error(err, "failed to disable crio profiling")
+			errs = fmt.Errorf("%w: %s", errs, err.Error())
+		}
+		if deleted {
+			r.EventRecorder.Eventf(r.CtrlConfig, corev1.EventTypeNormal, "DeleteConfig", "successfully deleted crio machine config")
+			r.PrevSyncChange["crio"] = PrevSyncData{
+				action: "deleted",
+			}
+		}
+	}
+
+	if r.CtrlConfig.Spec.EnableKubeletProfiling {
+		kubeletmc, created, err := r.ensureKubeletProfConfigExists(ctx)
+		if err != nil {
+			errored = true
+			r.Log.Error(err, "failed to enable kubelet profiling")
+			errs = fmt.Errorf("%w: %s", errs, err.Error())
+		}
+		if created {
+			r.EventRecorder.Eventf(r.CtrlConfig, corev1.EventTypeNormal, "CreateConfig", "successfully created kubelet config")
+			r.PrevSyncChange["kubelet"] = PrevSyncData{
+				action: "created",
+				config: *kubeletmc,
+			}
+		}
+	} else {
+		deleted, err := r.ensureKubeletProfConfigNotExists(ctx)
+		if err != nil {
+			errored = true
+			r.Log.Error(err, "failed to disable kubelet profiling")
+			errs = fmt.Errorf("%w: %s", errs, err.Error())
+		}
+		if deleted {
+			r.EventRecorder.Eventf(r.CtrlConfig, corev1.EventTypeNormal, "DeleteConfig", "successfully deleted kubelet config")
+			r.PrevSyncChange["kubelet"] = PrevSyncData{
+				action: "deleted",
+			}
+		}
+	}
+
+	if errored {
+		return errs
+	}
+	return nil
+}
+
+// checkMCPUpdateStatus is for reconciling update status of all machines in profiling MCP
+func (r *MachineconfigReconciler) checkMCPUpdateStatus(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	mcp := &mcv1.MachineConfigPool{}
+	key := types.NamespacedName{Namespace: r.CtrlConfig.Namespace, Name: ProfilingMCPName}
+	if err := r.Client.Get(ctx, key, mcp); err != nil {
+		r.Log.Error(err, "failed to fetch profiling MCP: %v", err)
+		return ctrl.Result{}, err
+	}
+
+	if mcv1.IsMachineConfigPoolConditionTrue(mcp.Status.Conditions, mcv1.MachineConfigPoolUpdating) &&
+		r.CtrlConfig.Status.UpdateStatus.InProgress == "false" {
+		r.EventRecorder.Eventf(r.CtrlConfig, corev1.EventTypeNormal, "ConfigUpdate", "config update still under progress")
```

Comment:
I would not throw an event here to bloat the event queue.

Reply:
comment incorporated in [6e77c7b](https://github.com/openshift/node-observability-operator/pull/16/commits/6e77c7b3116158c01e2e14500d27c633e7997885)